File: BOJ/C++/done/1016.cpp
#include <iostream>
#include <vector>
using namespace std;
long long mi, ma;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    int ans = 0;
    cin >> mi >> ma;
    vector<bool> visited(ma - mi + 1, false);
    for (long long i = 2; i * i <= ma; i++)
    {
        long long squre = i * i;
        long long start = (mi / squre) * squre;
        start = start < mi ? start + squre : start;
        for (long long j = start; j <= ma; j += squre)
        {
            visited[j - mi] = true;
        }
    }
    for (int i = 0; i < visited.size(); i++)
    {
        if (!visited[i])
            ans++;
    }
    cout << ans;
    return 0;
}

File: BOJ/JS/done/1016.js
const fs = require('fs');
const input = fs.readFileSync('./input.txt').toString().trim().split(' ');
const [min, max] = input.map(Number);
const visited = Array(max - min + 1).fill(false);

for (let i = 2; i * i <= max; i++) {
  const squre = i * i;
  let start = Math.ceil(min / squre) * squre;
  for (let j = start; j <= max; j += squre) {
    visited[j - min] = true;
  }
}

let ans = 0;
visited.forEach((v) => {
  if (!v) ans++;
});

console.log(ans);
